pontoon/base/views.py

```diff
@@ -294,19 +300,36 @@ def get_translations_from_other_locales(request):
         }, status=400)
 
     entity = get_object_or_404(Entity, pk=entity)
-    locales = entity.resource.project.locales.exclude(code=locale)
+    locale = get_object_or_404(Locale, code=locale)
+    selected_locales = request.user.profile.sorted_locales
     plural_form = None if entity.string_plural == "" else 0
 
+    preferred_source_translations = Translation.objects.filter(
+        entity=entity,
+        locale__in=selected_locales,
+        plural_form=plural_form,
+        approved=True
+    ).exclude(locale=locale)
```

Comment:
If you move this block under `translations =`, you can avoid code duplication and do:
```python
preferred = translations.filter(locale__in=preferred_locales)
other = translations.exclude(locale__in=preferred_locales)
```